lib/browser/navigation.js

```diff
@@ -70,6 +70,28 @@ function urlAsserter(url, query) {
   return new Asserter(getAndCheckUrl);
 }
 
+/**
+ * Wait for document to be in a certain state
+ * https://github.com/admc/wd#waiting-for-something
+ * https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState
+ *
+ * @param {object} self
+ * @param {string} state
+ * @param {number} [timeout] - Timeout for js condition to complete (optional)
+ */
+function waitForDocumentState(self, state, timeout) {
+  timeout = timeout || 8000;
+  var expr;
+  if (state === 'interactive') {
```

Comment:
Might be a good candidate for a switch statement (that would also be a good opportunity to `throw` in the `default:` branch).